Ft031 utworzenie modali

lkrzysztoff opened the pull request:
https://ft031-utworzenie-modali--meetek-envelo.netlify.app/theme

lkrzysztoff committed:
feat: added header component

this commit adds header and initials button which opens logout modal

lkrzysztoff committed:
feat: changed styles header component

changed margin between notifications button and initials button

lkrzysztoff committed:
feat: changed theme component

added h2 element above header in theme.component

lkrzysztoff committed:
fix: changed components logic and removed unnecessary code

this commit changed components logic, removed spaces in scss files and changed file name

lkrzysztoff committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/meetek-front into FT031-utworzenie-modali

lkrzysztoff committed:
feat: added modals

this commit add two modals that can take data and pass data

lkrzysztoff committed:
fix: added closeDialog function

this commit add closeDialog function in cancel confirm dialog

lkrzysztoff committed:
feat: added modals

this commit add two modals that can pass data

lkrzysztoff committed:
feat: added modals

this commit add two modals that can pass data

EMEZYK commented on src/app/core/theme.component.ts:
```diff
@@ -1,14 +1,16 @@
   template: `
     <h1>Storybook-like route</h1>
+    <h3>Dialogi</h3>
+    <app-example-dialog></app-example-dialog>
```
to tutaj w theme powinienes wywolac te 2 dialogi, oddzielnie input z dialogiemm, oddzielnie d ialogz buttonami; czyli logikę z example-dialog, przeniesc tutaj, a tamten folder wywalic

EMEZYK commented on src/app/shared/ui/modals/example dialog/example-dialog.component.ts:
```diff
@@ -0,0 +1,48 @@
+import { MatButtonModule } from '@angular/material/button';
+import { InputDialogComponent } from '../input dialog/input-dialog.component';
+import { CancelConfirmDialogComponent } from '../quit-section-dialog/cancel-confirm-dialog.component.ts';
+export interface DialogData {
+  title: string;
+  buttonText: string;
+  inputLabelText: string;
+  importedDialogData: string;
+}
```
interfejsy wynośmy do oddzielnego pliku

EMEZYK commented on src/app/shared/ui/modals/example dialog/example-dialog.component.ts:
```diff
@@ -0,0 +1,48 @@
+import { MatDialog } from '@angular/material/dialog';
+import { MatButtonModule } from '@angular/material/button';
+import { InputDialogComponent } from '../input dialog/input-dialog.component';
+import { CancelConfirmDialogComponent } from '../quit-section-dialog/cancel-confirm-dialog.component.ts';
```
dla kazdego folderu, zgodnie z nasza konwencja, tworzmy plik index.ts zeby skrocic sciezki importow

EMEZYK commented on src/app/shared/ui/modals/input dialog/input-dialog.component.html:
```diff
@@ -0,0 +1,25 @@
+  <div class="input-dialog__input-container">
+    <form class="example-form">
+      <input
+        class="input-dialog__input"
+        placeholder="{{data.inputLabelText}}"
+        [(ngModel)]="data.importedDialogData"
+        name="dialogData" />
```
dobra praktyka jest ustawiac rzeczy od wazniejszych do mniej waznych;  klasę przesunęłabym na koniec,a zaczelabym od ngModel

EMEZYK commented on src/app/shared/ui/modals/input dialog/input-dialog.component.ts:
```diff
@@ -0,0 +1,33 @@
+import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatButtonModule } from '@angular/material/button';
+import { MatCardModule } from '@angular/material/card';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
+import { MatInputModule } from '@angular/material/input';
+import { MatIconModule } from '@angular/material/icon';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { DialogData } from '../example dialog/example-dialog.component';
+import { FormsModule } from '@angular/forms';
```
importy zewnętrzne oddzielamy spacją od tyc naszych wewnątrznych

damiankubecki replied:
*enterem 😉

EMEZYK commented on src/app/shared/ui/modals/example dialog/example-dialog.component.html:
```diff
@@ -0,0 +1,3 @@
+<!-- Komponent stworzony na potrzeby wyśietlenia dialogów oraz przekazania do nich danych, w wersji produkcyjnej do usuniecia -->
+      <button mat-raised-button (click)="openDialog()">Dialog z inputem</button>
+      <button mat-raised-button (click)="openDialog2()">Dialog z przyciskami</button>
```
jeszcze jedna kwestia - jesli nazwa folderu sklada sie z wiecej niz 1 członu, to oddzielamjmy przez myslnik

damiankubecki commented on src/app/shared/ui/modals/quit-section-dialog/cancel-confirm-dialog.component.ts.ts:
```diff
@@ -0,0 +1,19 @@
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CancelConfirmDialogComponent {
+  constructor( public dialog:MatDialog, @Inject(MAT_DIALOG_DATA) public data: { text: string }) {}
```
to samo co wcześniej, czyli `inject` + `private` dla dialoga

damiankubecki commented on src/app/shared/ui/modals/quit-section-dialog/cancel-confirm-dialog.component.ts.ts:
```diff
@@ -0,0 +1,19 @@
+import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
+import { MatDialog, MatDialogModule, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+
+@Component({
+  selector: 'app-quit-section-dialog[text]',
+  standalone: true,
+  imports: [MatDialogModule, MatIconModule, MatButtonModule],
+  templateUrl: './cancel-confirm-dialog.component.html',
+  styleUrls: ['./cancel-confirm-dialog.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CancelConfirmDialogComponent {
+  constructor( public dialog:MatDialog, @Inject(MAT_DIALOG_DATA) public data: { text: string }) {}
+  closeDialog() {
+    this.dialog.closeAll()
+  }
+}
```
proszę użyć prettiera w tym pliku

damiankubecki commented on src/app/shared/ui/modals/quit-section-dialog/cancel-confirm-dialog.component.ts.ts:
```diff
@@ -0,0 +1,19 @@
+import { MatButtonModule } from '@angular/material/button';
+
+@Component({
+  selector: 'app-quit-section-dialog[text]',
```
czy używamy gdzieś `text`?

lkrzysztoff replied:
nie, to przypadkowo tam zostało

damiankubecki commented on src/app/shared/ui/modals/input dialog/input-dialog.component.html:
```diff
@@ -0,0 +1,25 @@
+    <form class="example-form">
+      <input
+        class="input-dialog__input"
+        placeholder="{{data.inputLabelText}}"
```
tutaj możemy użyć `[placeholder]="data.inputLabelText"`, wtedy od razu widać, że do `placeholder` przekazujemy wartość z ts

damiankubecki commented on src/app/shared/ui/modals/quit-section-dialog/cancel-confirm-dialog.component.html:
```diff
@@ -0,0 +1,14 @@
+    <p class="cancel-confirm-dialog__title">{{ data.text }}</p>
+  </div>
+  <div class="cancel-confirm-dialog__btn-container">
+    <button (click)="closeDialog()" mat-button class="cancel-confirm-dialog__cancel-btn error">ANULUJ</button>
+    <button mat-button class="cancel-confirm-dialog__confirm-btn success">POTWIERDŹ</button>
```
`text content` w tych buttonach powinien być napisany bez uppercase. Uppercase w naszych buttonach powinien być ustawiony w stylach globalnych (`text-transform: uppercase;`) - @jBachanek-02

damiankubecki commented on src/app/shared/ui/modals/input dialog/input-dialog.component.ts:
```diff
@@ -0,0 +1,33 @@
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class InputDialogComponent {
+  constructor(public dialog: MatDialog, @Inject(MAT_DIALOG_DATA) public data: DialogData) {}
```
przyjęliśmy w projekcie strategię wstrzykiwania zależności za pomocą `inject`, więc do wyrzucenia z `constructor` + dialog powinien być `private`

damiankubecki commented on src/app/shared/ui/modals/input dialog/input-dialog.component.scss:
```diff
@@ -0,0 +1,113 @@
+
+.input-dialog__close-btn {
+  background-color: var(--white);
+  color: var(--text);
```
zalecam dodanie `transition` - tak jak mają wszystkie buttony w naszej pięknej aplikacji

lkrzysztoff committed:
fix: deleted example dialog folder

deleted example dialog folder and moved its content to theme component

lkrzysztoff committed:
fix: deleted example dialog folder

deleted example dialog folder and moved its content to theme component

EMEZYK commented on src/app/shared/ui/modals/index.ts:
```diff
@@ -0,0 +1,3 @@
+export * from './input-dialog/dialog-data.interface';
+export * from './input-dialog/input-dialog.component';
+export * from './quit-section-dialog/cancel-confirm-dialog.component.ts';
```
ogólnie jeszcze jedna rzecz - ten folder i sam komponent nazwałabym 'confirm-dialog' , bo on nie tylko odnosi się do opuszczania sekcji, będziemy go reużywać

lkrzysztoff committed:
fix: renamed folder

this commit changed name of quit-section-dialog folder to cancel-confirm-dialog

lkrzysztoff committed:
fix: fixed input problems

this commit fix input problems and add validation to input

lkrzysztoff committed:
fix: removed useless import

this commit removed useless import

lkrzysztoff committed:
Merge branch 'develop' into FT031-utworzenie-modali
